remote-trait-object/src/port.rs

```diff
@@ -19,42 +19,83 @@ mod server;
 mod types;
 
 pub use self::types::Handler;
+use crate::forwarder::ServiceForwarder;
 use crate::ipc::multiplex::{MultiplexResult, Multiplexer};
 use crate::ipc::{IpcRecv, IpcSend};
+use crate::service::*;
+use client::Client;
+use server::Server;
+use std::sync::{Arc, Weak};
+
+pub trait Port: Send + Sync + 'static {
+    fn call(&self, arg: String) -> String;
+    fn delete_request(&self, id: ServiceObjectId);
+    fn register(&self, handle_to_register: Box<dyn Service>) -> ServiceObjectId;
+}
 
 pub struct BasicPort {
-    multiplexer: Option<Multiplexer>,
-    server: Option<server::Server>,
-    client: client::Client,
+    registry: Arc<ServiceForwarder>,
+    client: Client,
+}
+
+impl Port for BasicPort {
+    fn call(&self, arg: String) -> String {
+        self.client.call(arg)
+    }
+
+    fn delete_request(&self, _id: ServiceObjectId) {
+        unimplemented!()
+    }
+
+    fn register(&self, service: Box<dyn Service>) -> ServiceObjectId {
+        // FIXME
+        self.registry.add_service("DUMMY".to_owned(), service);
```

Comment:
How about getting a service name as string in the register function for now?
We can replace the string service name to ServiceObjectId later.

Reply:
@majecty Current code doesn't use return value of `register`, but just use reserved name `DUMMY`, because even if we return `DUMMY`, there is no way to deliver that to other moudle